### PR TITLE
Added entity view count cache module

### PR DESCRIPTION
**Pull request by LennardWesterveld**
Hello i added caching functionality because we had over 20.000 records and the query's where slow.
So we using now this table for sorting and displaying but doen't lose the functionality to limit the views by ip.

### CODE REVIEW

File: modules/entity_view_count_cache/entity_view_count_cache.admin.inc [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+  return $form;
+}
+
+/**
+ * Submit reindex form
+ */
+function entity_view_count_cache_reindex_form_submit() {
+  entity_view_count_cache_update_batch();
+}
```
Comment (RoySegall) [inline modules/entity_view_count_cache/entity_view_count_cache.admin.inc]: Missing line in the end of the file.

File: modules/entity_view_count_cache/entity_view_count_cache.module [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+function entity_view_count_cache_entity_insert($entity, $type) {
+  if ($type == 'entity_view_count' && $entity->type == 'entity_view_count_base') {
+
+    $cache_item = entity_view_count_cache_get($entity->entity_type, $entity->entity_id, $entity->entity_owner);
+    $count = entity_view_count_get_entity_count($entity->entity_type, $entity->entity_id);
+
+    drupal_alter('entity_view_count_cache', $count, $entity->entity_type, $entity->entity_id);
+
+    if (!empty($cache_item)) {
+      //update
```
Comment (RoySegall) [inline modules/entity_view_count_cache/entity_view_count_cache.module]: `//update` => `// Update.`

File: modules/entity_view_count_cache/entity_view_count_cache.module [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+    else {
+      entity_view_count_cache_insert($entity->entity_type, $entity->entity_id, $entity->entity_owner, $count);
+    }
+
+  }
+}
+
+/**
+ * Update cache item
+ * @param $id
```
Comment (RoySegall) [inline modules/entity_view_count_cache/entity_view_count_cache.module:83]: Missing PHPDOC.

File: modules/entity_view_count_cache/entity_view_count_cache.module [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+function entity_view_count_cache_update($id, $count, $type = ENTITY_VIEW_COUNT_CACHE_DEFAULT_TYPE) {
+  $query = db_update('entity_view_count_cache');
+  $query->fields(array('updated' => time(), 'count' => $count));
+  $query->condition('id', $id);
+
+  $query->execute();
+}
+
+/**
+ * Insert new cache item
```
Comment (RoySegall) [inline modules/entity_view_count_cache/entity_view_count_cache.module:96]: Missing PHPDOC.

File: modules/entity_view_count_cache/views/entity_view_count_cache_sort_handler_view_count.inc [hunk 5]
```diff
@@ -0,0 +1,67 @@
+<?php
+
+class entity_view_count_cache_sort_handler_view_count extends views_handler_sort {
```
Comment (RoySegall) [inline modules/entity_view_count_cache/views/entity_view_count_cache_sort_handler_view_count.inc:3]: Missing doc for the file.

File: modules/entity_view_count_cache/views/entity_view_count_cache.views.inc [hunk 6]
```diff
@@ -0,0 +1,20 @@
+<?php
+
```
Comment (RoySegall) [inline modules/entity_view_count_cache/views/entity_view_count_cache.views.inc:2]: Missing doc for the file.

File: modules/entity_view_count_cache/entity_view_count_cache.module [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+
+/**
+ * Worker for the batch updater
+ * @param $data
+ * @param $sandbox
+ */
+function entity_view_count_cache_update_batch_worker($data, &$sandbox) {
+  $sandbox['message'] = t('Now processing @entity_type @entity_id', array(
+    '@entity_type' => $data['entity_type'],
+    '@entity_id' => $data['entity_id']
```
Comment (RoySegall) [inline modules/entity_view_count_cache/entity_view_count_cache.module:191]: Missing a comma in the end of the file.

File: modules/entity_view_count_cache/entity_view_count_cache.module [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+ * @param $result
+ * @param $sandbox
+ */
+function entity_view_count_cache_update_batch_finish($result, $sandbox) {
+  drupal_set_message(t('Succeed indexing entity view cache items'));
+}
+
+/**
+ * Worker for the batch updater
+ * @param $data
```
Comment (RoySegall) [inline modules/entity_view_count_cache/entity_view_count_cache.module:185]: Missing a PHPDOC.

File: modules/entity_view_count_cache/entity_view_count_cache.module [hunk 9]
```diff
@@ -0,0 +1,198 @@
...
+ * Update batch to insert all already existing view items
+ */
+function entity_view_count_cache_update_batch() {
+  $batch = array(
+    'operations' => array(),
+    'finished' => 'entity_view_count_cache_update_batch_finish',
+    'title' => t('Entity view count cache indexing'),
+    'init_message' => t('Starting caching entity view count...'),
+    'progress_message' => t('Processed @current out of @total.'),
+    'error_message' => t('Has encountered an error.')
```
Comment (RoySegall) [inline modules/entity_view_count_cache/entity_view_count_cache.module:153]: Missing comma in the end of the array.

**Comment by RoySegall** [comment]
Thank you for this lovely PR.

two questions:
1. Why the caching support is in external module? I think it's has it's own place in the module it self.
2. Try to install drush using compoer - https://github.com/openscholar/openscholar/blob/SCHOLAR-3.x/.travis.yml#L33-L34

**Comment by LennardWesterveld** [comment]
Hi @RoySegall,

Answer on nr 1. Because i wanted to create caching without changing to original module.
Answer on nr 2. If you look in travis the error is just that drush cannot be installed through pear, but i think  this must be a other issues and not related to my code.

Kind regards,
Lennard

PS: i will fix the phpdocs

**Pull request closed without merging**